src/orquestra/sdk/_base/_driver/_models.py

```diff
@@ -168,23 +168,40 @@ def to_ir(self) -> TaskRun:
         )
 
 
-class WorkflowRunResponse(pydantic.BaseModel):
+class MinimalWorkflowRunResponse(pydantic.BaseModel):
     """
     Implements:
         https://github.com/zapatacomputing/workflow-driver/blob/main/openapi/src/schemas/WorkflowRun.yaml#L1
     """
 
     id: WorkflowRunID
     status: RunStatusResponse
+    definitionId: WorkflowDefID
+
+    def to_ir(self, workflow_def: WorkflowDef) -> WorkflowRun:
+        return WorkflowRun(
+            id=self.id,
+            status=self.status.to_ir(),
+            task_runs=[],
+            workflow_def=workflow_def,
+        )
```

Comment:
**issue**: later on, in a different part of the codebase we might see weird behavior when we have a variable of type `WorkflowRun` with always-empty list of task runs inside of it. In the past, we used [WorkflowRunMinimal](https://github.com/zapatacomputing/orquestra-workflow-sdk/blob/bc81bde8270ffe533c893d2ebca8569c26b7c43b/src/orquestra/sdk/schema/workflow_run.py#L51) for situations where we don't have some of the details. Did you consider returning the `*Minimal` object here?

I know `WorkflowRunMinimal` doesn't have a `workflow_def` field, but is that a problem? Do we ever need wf def from `WorkflowRunMinimal`?

Reply:
>  If we want to build the public API version of WorkflowRun, we need a workflow definition.

Could post an example of the public API where the `MinimalWorkflowRun` with wf def inside would be useful?

Reply:
Thanks. 
* `_api.WorkflowRun.__init__` and `_api.TaskRun.__init__` indeed only need the wf_run_id and wf_def, but not other fields from `WorkflowRun` model like the wf status, and all the task run details. 
* If we don't wanna populate all the `workflow_run.WorkflowRun`'s in the objects returned from `sdk.list_workflow_runs()` we shouldn't mark the output type as list of `workflow_run.WorkflowRun`. It's like marking a variable type as a non-optional `str` and setting it to `""` or `"empty"`.

Sounds like we need something `workflow_run.WorkflowRunMinimal` with a `wf_def` field. How about third type, `workflow_run.WorkflowRunSummary`?

Reply:
Yeah and yeah. I love the idea with a protocol, this will make it more open ended for the future.

Reply:
The link points to `.iter_logs()` instead of `.list_workflow_runs()` but I'm assuming it was a misclick

Reply:
@alexjuda Protocol would not play nice with the type checker and instead of spending a bunch of time and trying to get it to work I just used inheritance.